deadbelly/breathEZ
# Bje/feature/convert to pwa

### What does this PR do? 
The branch converts BreathEZ into a fully functional PWA. This includes the following: 

- A service worker, and appropriate calls to register that worker so that the app is installable.
- A js file that handles responses when the app is run offline. 
- An icon, which is apple touch compatible, and a maskable_icon. 
- A message that informs users of how to enable JS if they try to use the app with JS disabled, or in an incompatible environment. 

### Is this a feature or Fix/Refactor?  
Feature.

### Problems Encountered & Resulting Solutions  
Our path to running the service worker deviated from the boiler plate but we figured it out. Other than that this update was very straightforward. 

### Where should the reviewer start?
Look first at the new files that were added, and then at everything in the public directory. We did some of this together, but I know my understanding was shaky on Friday morning. I've spent a lot of time trying to understand what each files does so please ask me questions if you're confused by anything. I'm happy to get on a zoom call. 

### How should this be tested?
Install the app using the url path bar in the browser. You should then be able to close the app and re-open it natively on your device. Other than that no functionality was added, so it should continue to pass all unit and integration tests. 

### Steps To Take
We still need to make sure the app converts HTTP to HTTPS. This is a big deal for PWAs since installing apps natively raises security concerns. This will be dependent on our deployment environment. 

### This PR is related to which issues:
closes #39 closes #11
